plugins/outputs/sumologic/sumologic.go

```diff
@@ -0,0 +1,305 @@
+
+  ## Data format to be used for sending metrics.
+  ## This will set the "Content-Type" header accordingly.
+  ## Currently supported formats: 
+  ## * graphite - for Content-Type of application/vnd.sumologic.graphite
+  ## * carbon2 - for Content-Type of application/vnd.sumologic.carbon2
+  ## * prometheus - for Content-Type of application/vnd.sumologic.prometheus
+  ##
+  ## More information can be found at:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#content-type-headers-for-metrics
+  ##
+  ## NOTE:
+  ## When unset, telegraf will by default use the influx serializer which is currently unsupported
+  ## in HTTP Source.
+  data_format = "carbon2"
+
+  ## Timeout used for HTTP request
+  # timeout = "5s"
+  
+  ## HTTP method, one of: "POST" or "PUT". "POST" is used by default if unset.
+  # method = "POST"
+
+  ## Max HTTP request body size in bytes before compression (if applied).
+  ## By default 1MB is recommended.
+  ## NOTE:
+  ## Bear in mind that in some serializer a metric even though serialized to multiple
+  ## lines cannot be split any further so setting this very low might not work
+  ## as expected.
+  # max_request_body_size = 1_000_000
+
+  ## Additional, Sumo specific options.
+  ## Full list can be found here:
+  ## https://help.sumologic.com/03Send-Data/Sources/02Sources-for-Hosted-Collectors/HTTP-Source/Upload-Metrics-to-an-HTTP-Source#supported-http-headers
+
+  ## Desired source name.
+  ## Useful if you want to override the source name configured for the source.
+  # source_name = ""
+
+  ## Desired host name.
+  ## Useful if you want to override the source host configured for the source.
+  # source_host = ""
+
+  ## Desired source category.
+  ## Useful if you want to override the source category configured for the source.
+  # source_category = ""
+
+  ## Comma-separated key=value list of dimensions to apply to every metric.
+  ## Custom dimensions will allow you to query your metrics at a more granular level.
+  # dimensions = ""
+`
+
+	defaultClientTimeout      = 5 * time.Second
+	defaultMethod             = http.MethodPost
+	defaultMaxRequestBodySize = 1_000_000
+
+	contentTypeHeader     = "Content-Type"
+	carbon2ContentType    = "application/vnd.sumologic.carbon2"
+	graphiteContentType   = "application/vnd.sumologic.graphite"
+	prometheusContentType = "application/vnd.sumologic.prometheus"
+)
+
+type header string
+
+const (
+	sourceNameHeader     header = `X-Sumo-Name`
+	sourceHostHeader     header = `X-Sumo-Host`
+	sourceCategoryHeader header = `X-Sumo-Category`
+	dimensionsHeader     header = `X-Sumo-Dimensions`
+)
+
+type SumoLogic struct {
+	URL               string            `toml:"url"`
+	Timeout           internal.Duration `toml:"timeout"`
+	Method            string            `toml:"method"`
+	MaxRequstBodySize int               `toml:"max_request_body_size"`
```

Comment:
There is a config.Size type that might be nice for this.

Reply:
Oh, I didn't see this. Nice! Changed